src/cs50.h

```diff
@@ -52,17 +52,45 @@ typedef char *string;
 /**
  * Temporarily used to make arguments to get_* (but not Get*) optional.
  */
-struct prompt
-{
-    int _sentinel;
-    string prompt;
-};
-
-/** 
- * Some compilers warn about the trick we use for optional arguments. This
- * line overrides this warning.
- */
-#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+
+#define CONCAT(a, b) a ## b 
+#define SECOND(first, second, ...) second
+#define NOT_(...) SECOND(__VA_ARGS__, 0)
+#define NOT_SENTINEL_0 ignore_me, 1
+#define NOT(cond) NOT_(CONCAT(NOT_SENTINEL_, cond))
+#define BOOL(cond) NOT(NOT(cond))
+
+#define IF_ELSE(cond) IF_ELSE_(BOOL(cond))
+#define IF_ELSE_(cond) CONCAT(IF_, cond)
+#define IF_1(...) __VA_ARGS__ IF_1_ELSE
+#define IF_0(...) IF_0_ELSE
+#define IF_1_ELSE(...)
+#define IF_0_ELSE(...) __VA_ARGS__
+
+#define ARG(_0, _1, _2, _3, _4, _5, _6, _7, _8,             \
+            _9, _10, _11, _12, _13, _14, _15, _16,          \
+            _17, _18, _19, _20, _21, _22, _23, _24,         \
+            _25, _26, _27, _28, _29, _30, _31, ...) _31
```

Comment:
Albeit extreme, do you want to support 127 for apparent standard's sake? Cf. https://gcc.gnu.org/onlinedocs/gcc-4.3.5/cpp/Implementation-limits.html.